ompi/mpi/man/man3/MPI_Gather.md

```diff
@@ -0,0 +1,205 @@
+```
+
+## Fortran Syntax
+
+```fortran
+USE MPI
+! or the older form: INCLUDE 'mpif.h'
+
+MPI_GATHER(SENDBUF, SENDCOUNT, SENDTYPE, RECVBUF, RECVCOUNT,
+    	RECVTYPE, ROOT, COMM, IERROR)
+    <type>	SENDBUF(*), RECVBUF(*)
+    INTEGER	SENDCOUNT, SENDTYPE, RECVCOUNT, RECVTYPE, ROOT
+    INTEGER	COMM, IERROR
+
+MPI_IGATHER(SENDBUF, SENDCOUNT, SENDTYPE, RECVBUF, RECVCOUNT,
+    	RECVTYPE, ROOT, COMM, REQUEST, IERROR)
+    <type>	SENDBUF(*), RECVBUF(*)
+    INTEGER	SENDCOUNT, SENDTYPE, RECVCOUNT, RECVTYPE, ROOT
+    INTEGER	COMM, REQUEST, IERROR
+```
+
+## Fortran 2008 Syntax
+
+```fortran
+USE mpi_f08
+
+MPI_Gather(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype,
+    	root, comm, ierror)
+    TYPE(*), DIMENSION(..), INTENT(IN) :: sendbuf
+    TYPE(*), DIMENSION(..) :: recvbuf
+    INTEGER, INTENT(IN) :: sendcount, recvcount, root
+    TYPE(MPI_Datatype), INTENT(IN) :: sendtype, recvtype
+    TYPE(MPI_Comm), INTENT(IN) :: comm
+    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
+
+MPI_Igather(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype,
+    	root, comm, request, ierror)
+    TYPE(*), DIMENSION(..), INTENT(IN), ASYNCHRONOUS :: sendbuf
+    TYPE(*), DIMENSION(..), ASYNCHRONOUS :: recvbuf
+    INTEGER, INTENT(IN) :: sendcount, recvcount, root
+    TYPE(MPI_Datatype), INTENT(IN) :: sendtype, recvtype
+    TYPE(MPI_Comm), INTENT(IN) :: comm
+    TYPE(MPI_Request), INTENT(OUT) :: request
+    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
+```
+
+
+# Input Parameters
+
+* `sendbuf` : Starting address of send buffer (choice).
+* `sendcount` : Number of elements in send buffer (integer).
+* `sendtype` : Datatype of send buffer elements (handle).
+* `recvcount` : Number of elements for any single receive (integer, significant only
+at root).
+* `recvtype` : Datatype of recvbuffer elements (handle, significant only at root).
+* `root` : Rank of receiving process (integer).
+* `comm` : Communicator (handle).
+
+
+# Output Parameters
+
+* `recvbuf` : Address of receive buffer (choice, significant only at root).
+* `request` : Request (handle, non-blocking only).
+* `IERROR` : Fortran only: Error status (integer).
+
+# Description
+
+Each process (root process included) sends the contents of its send
+buffer to the root process. The root process receives the messages and
+stores them in rank order. The outcome is as if each of the n processes
+in the group (including the root process) had executed a call to
+
+```c
+MPI_Send(sendbuf, sendcount, sendtype, root, ...)
+```
+
+and the root had executed n calls to
+
+```c
+MPI_Recv(recfbuf + i * recvcount * extent(recvtype), recvcount, recvtype, i, ...)
+```
+
+where extent(recvtype) is the type extent obtained from a call to
+`MPI_Type_extent()`.
+
+An alternative description is that the n messages sent by the processes
+in the group are concatenated in rank order, and the resulting message
+is received by the root as if by a call to `MPI_RECV(recvbuf, recvcount
+* n, recvtype, ... )`.
```

Comment:
Something went wrong here -- when you look at the rendered version, it doesn't look right.  Is the `*n` supposed to be there?  That might be throwing off the Markdown rendering (i.e., the Markdown renderer might be getting confused by a as the first character and choosing to render it as a bullet).

Reply:
Got it! I moved the * to line 106 right after recvcount and keep "n, recvtype, ...) in line 107.